backends/tfhe-cuda-backend/cuda/include/integer/integer_utilities.h

```diff
@@ -20,6 +20,32 @@ class NoiseLevel {
   static const uint64_t UNKNOWN = std::numeric_limits<uint64_t>::max();
 };
 
+#ifdef DEBUG
+#define CHECK_NOISE_LEVEL(noise_level_expr, msg_mod, carry_mod)                \
+  do {                                                                         \
+    if ((msg_mod) == 2 && (carry_mod) == 2) {                                  \
+      constexpr int max_noise_level = 3;                                       \
+      if ((noise_level_expr) > max_noise_level)                                \
+        PANIC("Cuda error: noise exceeds maximum authorized value for 1_1 "    \
+              "parameters");                                                   \
+    } else if ((msg_mod) == 4 && (carry_mod) == 4) {                           \
+      constexpr int max_noise_level = 5;                                       \
+      if ((noise_level_expr) > max_noise_level)                                \
+        PANIC("Cuda error: noise exceeds maximum authorized value for 2_2 "    \
+              "parameters");                                                   \
+    } else if ((msg_mod) == 8 && (carry_mod) == 8) {                           \
+      constexpr int max_noise_level = 9;                                       \
+      if ((noise_level_expr) > max_noise_level)                                \
+        PANIC("Cuda error: noise exceeds maximum authorized value for 3_3 "    \
+              "parameters");                                                   \
+    }                                                                          \
```

Comment:
there are cases when this macro is called on 0, 0 for message modulus and carry modulus and we could want to prevent invalid values being passed too:

maybe add:
```
else if ((msg_mod) == 0 && (carry_mod) ==0) {
  // ok
} else {
 assert(false, "Invalid message modulus or carry modululs")
}
```

Reply:
Ah yes sorry that's what I did for shortint level ops where message & carry modulus are not actually defined.